test/test_torch.py

```diff
@@ -6397,80 +6398,210 @@ def test_logical_and(self, device):
 
-        out = m1.clone()
+        out = torch.empty_like(m1)
         torch.clamp(m1, min=min_val, max=max_val, out=out)
-        self.assertEqual(out, res1)
+        self.assertEqual(m1_base, out)
 
-        res1 = torch.clamp(m1, min=min_val)
-        res2 = m1.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(min_val, res2[i])
-        self.assertEqual(res1, res2)
+        m1.clamp_(min_val, max_val)
+        self.assertEqual(m1_base, m1)
+
+    def test_clamp_min(self, device):
+        min_val = -1
+        m1, m1_base = self.generate_clamp_baseline(device, min_vals=min_val)
 
+        m1_res = torch.clamp(m1, min=min_val)
+        self.assertEqual(m1_base, m1_res)
+
+        out = torch.empty_like(m1)
         torch.clamp(m1, min=min_val, out=out)
-        self.assertEqual(out, res1)
+        self.assertEqual(m1_base, out)
 
-        res1 = torch.clamp(m1, max=max_val)
-        res2 = m1.clone()
-        for i in iter_indices(res2):
-            res2[i] = min(max_val, res2[i])
-        self.assertEqual(res1, res2)
+    def test_clamp_max(self, device):
+        max_val = 1
+        m1, m1_base = self.generate_clamp_baseline(device, max_vals=max_val)
+
+        m1_res = torch.clamp(m1, max=max_val)
+        self.assertEqual(m1_base, m1_res)
 
+        out = torch.empty_like(m1)
         torch.clamp(m1, max=max_val, out=out)
-        self.assertEqual(out, res1)
+        self.assertEqual(m1_base, out)
 
-        # if the tensor contains nan case
-        test_tens = torch.tensor([nan], device=device)
+    def test_clamp_propagates_nans(self, device):
+        min_val = -1
+        max_val = 1
+        m1, m1_base = self.generate_clamp_baseline(device, min_val, max_val, with_nans=True)
 
-        res1 = test_tens.clone()
-        res1.clamp_(min_val, max_val)
-        res2 = test_tens.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(min(res2[i], max_val), min_val)
-        self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+        m1_res = torch.clamp(m1, min_val, max_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1_res))
 
-        out = test_tens.clone()
-        torch.clamp(test_tens, min=min_val, max=max_val, out=out)
-        self.assertEqual(torch.isnan(out), torch.isnan(res1))
+        out = torch.empty_like(m1)
+        torch.clamp(m1, min_val, max_val, out=out)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(out))
 
-        res1 = torch.clamp(test_tens, min=min_val)
-        res2 = test_tens.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(res2[i], min_val)
-        self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+        m1.clamp_(min_val, max_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1))
 
-        torch.clamp(test_tens, min=min_val, out=out)
-        self.assertEqual(torch.isnan(out), torch.isnan(res1))
+    def test_clamp_min_propagates_nans(self, device):
```

Comment:
You can probably roll this test into the previous test_clamp_propagates_nans by taking the product of min/max args and not passing those args to the function if they're None.